src/Uno.UI/UI/Xaml/WindowManagerInterop.wasm.cs

```diff
@@ -496,6 +496,124 @@ private struct WindowManagerSetStylesParams
 
 		#endregion
 
+		#region SetSolidColorBorder
+
+		internal static void SetSolidColorBorder(IntPtr htmlId, Color color, string borderWidth)
+		{
+			if (UseJavascriptEval)
+			{
+				var command = "Uno.UI.WindowManager.current.setSolidColorBorder(" + htmlId + ", \"" + color.ToCssInteger() + "\", \"" + WebAssemblyRuntime.EscapeJs(borderWidth) + "\")";
+
+				WebAssemblyRuntime.InvokeJS(command);
+			}
+			else
+			{
+				var parms = new WindowManagerSetSolidColorBorderParams
+				{
+					HtmlId = htmlId,
+					Color = color.ToCssInteger(),
+					Width = borderWidth,
+				};
+
+				TSInteropMarshaller.InvokeJS("Uno:setSolidColorBorderNative", parms);
+			}
+		}
+
+		[TSInteropMessage]
+		[StructLayout(LayoutKind.Sequential, Pack = 4)]
+		private struct WindowManagerSetSolidColorBorderParams
+		{
+			public IntPtr HtmlId;
+
+			public uint Color;
+
+			[MarshalAs(TSInteropMarshaller.LPUTF8Str)]
+			public string Width;
+		}
+
+		#endregion
+
+		#region SetGradientBorder
+
+		internal static void SetGradientBorder(IntPtr htmlId, string borderImage, string borderWidth)
+		{
+			if (UseJavascriptEval)
+			{
+				var command = "Uno.UI.WindowManager.current.setGradientBorder(" + htmlId + ", \"" + WebAssemblyRuntime.EscapeJs(borderImage) + "\", \"" + WebAssemblyRuntime.EscapeJs(borderWidth) + "\")";
+
+				WebAssemblyRuntime.InvokeJS(command);
+			}
+			else
+			{
+				var parms = new WindowManagerSetGradientBorderParams
+				{
+					HtmlId = htmlId,
+					BorderImage = borderImage,
+					Width = borderWidth,
+				};
+
+				TSInteropMarshaller.InvokeJS("Uno:setGradientBorderNative", parms);
+			}
+		}
+
+		[TSInteropMessage]
+		[StructLayout(LayoutKind.Sequential, Pack = 4)]
+		private struct WindowManagerSetGradientBorderParams
+		{
+			public IntPtr HtmlId;
+
+			[MarshalAs(TSInteropMarshaller.LPUTF8Str)]
+			public string BorderImage;
+
+			[MarshalAs(TSInteropMarshaller.LPUTF8Str)]
+			public string Width;
+		}
+
+		#endregion
+
+		#region SetBorderRadius
+
+		internal static void SetBorderRadius(IntPtr htmlId, CornerRadius cornerRadius)
+		{
+			if (UseJavascriptEval)
+			{
+				var topLeft = cornerRadius.TopLeft.ToStringInvariant();
+				var topRight = cornerRadius.TopRight.ToStringInvariant();
+				var bottomLeft = cornerRadius.BottomLeft.ToStringInvariant();
+				var bottomRight = cornerRadius.BottomRight.ToStringInvariant();
+				var command = "Uno.UI.WindowManager.current.setBorderRadius(" + htmlId + ", \"" + topLeft + "\",\"" + topRight + "\",\"" + bottomRight + "\",\"" + bottomLeft + "\")";
```

Comment:
we can add a string extensions for double-quoting
it would make this command easier to write/read
or
we could just use the """ approach